backend/qemu.pm

```diff
@@ -346,6 +347,55 @@ sub deflate_balloon ($self) {
     $self->handle_qmp_command({execute => 'balloon', arguments => {value => $ram_bytes}}, fatal => 1);
 }
 
+sub save_storage ($self, $args) {
+    my $vars = \%bmwqemu::vars;
+    my $bdc = $self->{proc}->blockdev_conf;
+    my $fname = $args->{filename};
+    my $rsp = $self->handle_qmp_command({execute => 'query-status'}, fatal => 1);
+    bmwqemu::diag("Saving storage devices (current VM state is $rsp->{return}->{status})");
+    my $was_running = $rsp->{return}->{status} eq 'running';
+    if ($was_running) {
+        $self->inflate_balloon();
+        $self->freeze_vm();
+    }
+    mkpath("assets_public");
+    $bdc->for_each_drive(sub ($drive) {
+            my $size = $drive->{drive}->{size};
+            my $id = "$drive->{id}-backup-$fname";
+            my $node = $drive->{drive}->{node_name};
+            # no need to save CDs
+            return if ($node =~ qr/cd[0-9]-overlay/);
+            my $my_node = "$node-$fname";
+            my $bck_file = "assets_public/$my_node-$vars->{NAME}.qcow2";
+            # create disk
+            runcmd('qemu-img', 'create', '-f', 'qcow2', "$bck_file", $size);
```

Comment:
Why does `$bck_file` need quotes?

Reply:
filepath .. so can contain space and other awful characters...

Reply:
Sure it can. But the quotes around the variable don't make any difference.
We're using perl here, not bash.

Reply:
Look at what happens here:
```
% perl -wE'
sub runcmd {
  my ($path) = @_;
  say "path: >>$path<<";
}
my $arg = "some path with spaces";
runcmd($arg);
runcmd("$arg");'
path: >>some path with spaces<<
path: >>some path with spaces<<
```
So in the function `runcmd` it will end up without quotes in any case.

If you wanted to add quotes, you would have to do `qq{"$bck_file|}`. But that doesn't make sense. The `runcmd` function will take care of it anyway. Those variables aren't passed to a shell. Otherwise we would already have a problem with existing `runcmd` calls.